bundles/framework/mydata/instance.js

```diff
@@ -0,0 +1,329 @@
+         * Sets the sandbox reference to this component
+         */
+        setSandbox: function (sandbox) {
+            this.sandbox = sandbox;
+        },
+        /**
+         * @method getSandbox
+         * @return {Oskari.Sandbox}
+         */
+        getSandbox: function () {
+            return this.sandbox;
+        },
+        /**
+         * @method getViewService
+         * @return {Oskari.mapframework.bundle.personaldata.service.ViewService}
+         */
+        getViewService: function () {
+            return this.viewService;
+        },
+        /**
+         * @method getLocalization
+         * Returns JSON presentation of bundles localization data for current language.
+         * If key-parameter is not given, returns the whole localization data.
+         *
+         * @param {String} key (optional) if given, returns the value for key
+         * @return {String/Object} returns single localization string or
+         *      JSON object for complete data depending on localization
+         *      structure and if parameter key is given
+         */
+        getLocalization: function (key) {
+            if (!this._localization) {
+                this._localization = Oskari.getLocalization(this.getName());
+            }
+            if (key) {
+                return this._localization[key];
+            }
+            return this._localization;
+        },
+        /**
+         * @method start
+         * implements BundleInstance protocol start methdod
+         */
+        'start': function () {
+            var me = this;
+
+            if (me.started) {
+                return;
+            }
+
+            me.started = true;
+
+            var conf = this.conf || {};
+            this.conf = conf;
+            var sandboxName = (conf ? conf.sandbox : null) || 'sandbox';
+            var sandbox = Oskari.getSandbox(sandboxName),
+                p;
+
+            me.sandbox = sandbox;
+            this.viewService = Oskari.clazz.create('Oskari.mapframework.bundle.personaldata.service.ViewService', Oskari.urls.getRoute());
+
+            sandbox.register(me);
+            for (p in me.eventHandlers) {
+                if (me.eventHandlers.hasOwnProperty(p)) {
+                    sandbox.registerForEventByName(me, p);
+                }
+            }
+
+            // Let's extend UI
+            var request = Oskari.requestBuilder('userinterface.AddExtensionRequest')(this);
+
+            sandbox.request(this, request);
+
+            // draw ui
+            me.createUi();
+
+            // Flyout exists and we can hook up the request handler
+            this.requestHandlers = {
+                addTabRequestHandler: Oskari.clazz.create('Oskari.mapframework.bundle.personaldata.request.AddTabRequestHandler', sandbox, this.plugins['Oskari.userinterface.Flyout'])
+            };
+            sandbox.requestHandler('PersonalData.AddTabRequest', this.requestHandlers.addTabRequestHandler);
+
```

Comment:
Let's remove the request from the mydata bundle. The problem with the request is that everything in a request should be serializable to JSON and the current jQuery and mydata React refs + handlers are not.

```suggestion
            // Flyout exists and we can hook up the request handler
            this.requestHandlers = {
                addTabRequestHandler: Oskari.clazz.create('Oskari.mapframework.bundle.personaldata.request.AddTabRequestHandler', sandbox, this.plugins['Oskari.userinterface.Flyout'])
            };
            sandbox.requestHandler('PersonalData.AddTabRequest', this.requestHandlers.addTabRequestHandler);

```

Reply:
Instead lets add a service that is registered to sandbox. The service should provide a way for bundles to call the `addTab()` function. Other bundles could then register their tabs with `sandbox.getService('Oskari.mydata.MydataService').addTab()`